innovation.game.php

```diff
@@ -22829,7 +22886,28 @@ function stInterInteractionStep() {
                     self::executeDraw($player_id, $value_to_score, 'score');
                     self::executeDraw($player_id, $value_to_score, 'score');
                     break;     
-                
+                // id 413, Echoes age 8: Crossword
+                case "413N1A":
+                    $bonus_counts = self::getAuxiliaryArray();
+                    $remaining_values = count(array_filter($bonus_counts));
+                    // if more than 1 value remains, repeat the choice loop
+                    if($remaining_values > 1) {
+                        $step--;
+                        self::incrementStep(-1);
+                    } else {
+                        // if only one value remains, find the value and draw that many cards
+                        for($i = 0; i < $bonus_counts; $i++) {
+                            if($bonus_counts[$i] > 0) {
+                                for($d = 0; $d < $bonus_counts[$i]; $d++) {
+                                    self::executeDraw($player_id, $i + 1, 'hand');
+                                }
+                                break;
+                            }
+                        }
+                        $step++;
```

Comment:
You don't need to specifically advance the step as that happens automatically.  You can think of the incrementStep function as your only way to keep the step from advancing.

In rare occurrences, you might need to skip an interaction, but I don't think its necessary here.